homeassistant/components/hyperion/light.py

```diff
@@ -84,45 +67,58 @@
     }
 )
 
+ICON_LIGHTBULB = "mdi:lightbulb"
+ICON_EFFECT = "mdi:lava-lamp"
+ICON_EXTERNAL_SOURCE = "mdi:video-input-hdmi"
+
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up a Hyperion server remote."""
     name = config[CONF_NAME]
     host = config[CONF_HOST]
     port = config[CONF_PORT]
     priority = config[CONF_PRIORITY]
-    hdmi_priority = config[CONF_HDMI_PRIORITY]
-    default_color = config[CONF_DEFAULT_COLOR]
-    effect_list = config[CONF_EFFECT_LIST]
 
-    device = Hyperion(
-        name, host, port, priority, default_color, hdmi_priority, effect_list
-    )
+    # Use loop outside of HA's tracked task in order not to delay startup.
+    hyperion_client = client.HyperionClient(host, port, loop=hass.loop)
```

Comment:
The underlying client library uses `create_task` on that loop:

```
self._manage_connection_task = self._loop.create_task(manage(self))
```

Are you recommending I pass a callback in to create_task on that loop instead of passing the loop directly, or call create_task outside the library, or something else?

Reply:
As the library we be creating the task inside the event loop it can do so directly with `asyncio.create_task`. It doesn't need to access the loop directly at all.

If the library is only supporting Python 3.7+ I recommend to remove the loop parameter for the async client.

Reply:
Ah, it's just using get_event_loop() under the hood, got it! Thanks.